nearcore/src/runtime/mod.rs

```diff
@@ -500,6 +505,29 @@ impl NightshadeRuntime {
 
         Ok(result)
     }
+
+    fn precompile_contracts(
+        &self,
+        protocol_version: ProtocolVersion,
+        contract_codes: Vec<ContractCode>,
+    ) -> Result<(), Error> {
+        let runtime_config =
+            RuntimeConfig::from_protocol_version(&self.genesis_runtime_config, protocol_version);
+        let compiled_contract_cache: Option<Arc<dyn CompiledContractCache>> =
+            Some(Arc::new(StoreCompiledContractCache { store: self.store.clone() }));
+        contract_codes
+            .par_iter()
+            .map(|code| {
+                precompile_contract(
+                    &code,
+                    &runtime_config.wasm_config,
+                    compiled_contract_cache.as_deref(),
+                )
+                .ok();
+            })
+            .collect::<()>();
```

Comment:
What's the reason for not writing a loop here? The code here is hard to read.

Reply:
That's `par_iter()` -- rayon's parallel iterator. A for loop would be more readable, but sequential.

Reply:
Also, I changed `map + collect` to `for_each`.